client/template.go

```diff
@@ -128,6 +130,10 @@ func (payload *TemplateCreatePayload) Invalidate() error {
 		return errors.New("must supply opentofu version")
 	}
 
+	if payload.TerragruntTfBinary != "" && payload.Type != "terragrunt" {
```

Comment:
This is to avoid setting `TerragruntTfBinary` without a template of type `terragrunt`.